### PR TITLE
Change privilege of enrich stats API to monitor

### LINKED ISSUE #51677: 'enrich stats' API not included in 'monitor' security privilege
**Elasticsearch version**:
`Version: 7.5.2, Build: default/tar/8bec50e1e0ad29dad5653712cf3bb580cd1afcdf/2020-01-15T12:11:52.313576Z, JVM: 11.0.6`

**JVM version**:
`OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.6+10, mixed mode)`

**OS version**:
`Linux 3.10.0-1062.1.2.el7.x86_64 #1 SMP Mon Sep 16 14:19:51 EDT 2019 x86_64 x86_64 x86_64 GNU/Linux`

**Description of the problem including expected versus actual behavior**:
Using Metricbeat's `elasticsearch` module with `xpack.enabled: true` and the built-in`remote_monitoring_user` to monitor the Elasticsearch instance gives HTTP/403 errors for the `enrich` [metricset](https://www.elastic.co/guide/en/beats/metricbeat/current/metricbeat-metricset-elasticsearch-enrich.html).

It seems the `monitor` privilege does not cover the `enrich stats` [API](https://www.elastic.co/guide/en/elasticsearch/reference/current/enrich-stats-api.html), and therefore the `remote_monitoring_collector` role fails too.

As a workaround one can create a custom role with the `manage_enrich` privilege, but of course this is way too permissive just for monitoring.

**Provide logs (if relevant)**:
```
2020-01-30T14:55:24.941+0100	INFO	module/wrapper.go:252	Error fetching data for metricset elasticsearch.enrich: HTTP error 403 in : 403 Forbidden
```

### PR DESCRIPTION
**Pull request by ywangd**
The `remote_monitoring_user` user needs to access the enrich stats API.
But the request is denied because the API is categorized under `admin`.
The correct privilege should be `monitor`.

resolves: #51677

### CODE REVIEW

**Review by martijnvg — APPROVED** [review]
LGTM

**Comment by tvernum** [comment]
I see this is labeled `8.0.0` only, and not `7.7.0`. Is that intentional?

**Comment by ywangd** [comment]
No I just wasn’t sure whether backport is needed. It’s added now.

File: x-pack/qa/rolling-upgrade/src/test/resources/rest-api-spec/test/mixed_cluster/100_enrich.yml
Comment (ywangd) [inline x-pack/qa/rolling-upgrade/src/test/resources/rest-api-spec/test/mixed_cluster/100_enrich.yml]: @tvernum This new test [failed](https://gradle-enterprise.elastic.co/s/rjxpmmummomge/tests/qvsbxgij46xlg-gx4w5a2ll3kls) in a mixed cluster environment. Next steps are:

* Mute this test
* Merge this PR once approved
* Backport and fix the bug in 7.x and merge
* Another PR to unmute this test

But I feel it is a bit unwieldy just to stick to the backport workflow. Alternatively I could:
* Fix this bug and merge to 7.x (so 7.x will be the shim)
* Merge this PR

Please let me know your thoughts on this. Thanks!

Comment (tvernum) [reply x-pack/qa/rolling-upgrade/src/test/resources/rest-api-spec/test/mixed_cluster/100_enrich.yml]: My personal preference is to stick with the standard process of merge to master before backporting, but I don't feel so strongly as to stop you doing what makes sense to you.

Comment (ywangd) [reply x-pack/qa/rolling-upgrade/src/test/resources/rest-api-spec/test/mixed_cluster/100_enrich.yml]: I am happy to following the standard process. No strong opinion here. In fact, I have muted the test so it can be merged first. 
But should we get the other 7.x PR to a state where it is almost ready before merging this one? Because as soon as it is merged, a bug is introduced and it can only be fixed by merging the 7.x PR. I'd like to keep the merge time between the two PRs minimal so the bug will *not* be around for long.

**Comment by ywangd** [comment]
As suggested, added the [bwc API change](https://github.com/elastic/elasticsearch/pull/52027/files#diff-e3bdce9ba7ee7ea744f10e816c8a8806R139-R141) to `TransportNodesAction`.

@tvernum Could you please take another look? I'd like to get this PR into master first and then work on the 7.x backport to following the standard process.

**Review by tvernum — APPROVED** [review]
LGTM, Sorry for sitting on this for so long.

**Pull request merged**
